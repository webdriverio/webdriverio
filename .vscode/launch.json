{
  // Use IntelliSense to learn about possible attributes.
  // Hover to view descriptions of existing attributes.
  // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
  "version": "0.2.0",
  "configurations": [
    {
      "type": "node",
      "request": "launch",
      "name": "Run Linter Tests",
      "skipFiles": ["<node_internals>/**"],
      "cwd": "${workspaceRoot}",
      "console": "integratedTerminal",
      "runtimeExecutable": "npx",
      "runtimeArgs": ["run-s", "test:eslint", "test:ejslint"]
    }, {
      "type": "node",
      "request": "launch",
      "name": "Run Unit Tests",
      "skipFiles": ["<node_internals>/**"],
      "cwd": "${workspaceRoot}",
      "console": "integratedTerminal",
      "runtimeExecutable": "npm",
      "runtimeArgs": ["run-script", "test:unit:run"]
    }, {
      "type": "node",
      "request": "launch",
      "name": "Run Smoke Tests",
      "skipFiles": ["<node_internals>/**"],
      "cwd": "${workspaceRoot}",
      "console": "integratedTerminal",
      "runtimeExecutable": "npm",
      "runtimeArgs": ["run-script", "test:smoke"]
    }
  ]
}
