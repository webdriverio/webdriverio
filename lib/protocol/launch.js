/**
 *
 * Launch the session for the desired capabilities. Note that this is the companion
 * to the `autoLaunch=false` capability. This is not for launching arbitrary
 * apps/activities --- for that use [startActivity](/api/mobile/startActivity.html).
 * This is for continuing the initialization ("launch") process if you have used
 * `autoLaunch=false`.
 *
 * <example>
    :launch.js
    browser.launch()
 * </example>
 *
 * @see  https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/appium-bindings.md#launch
 * @type mobile
 * @for ios, android
 *
 */

let launch = function () {
    return this.requestHandler.create({
        path: '/session/:sessionId/appium/app/launch',
        method: 'POST'
    })
}

export default launch
