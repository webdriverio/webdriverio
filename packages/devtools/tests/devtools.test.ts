import launch from '../src/launcher'
import DevTools from '../src'

jest.mock('../src/launcher', () => jest.fn().mockImplementation((capabilities) => {
    capabilities['goog:chromeOptions'] = capabilities['goog:chromeOptions'] || {}
    return {
        pages: jest.fn().mockReturnValue(Promise.resolve([{
            on: jest.fn(),
            setDefaultTimeout: jest.fn()
        }])),
        _connection: {
            url: () => 'ws://localhost:49375/devtools/browser/c4b017ea-f476-4026-a699-bc5d4858cfe1'
        },
        userAgent: jest.fn().mockReturnValue(Promise.resolve('Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.149 Safari/537.36'))
    }
}))

beforeEach(() => {
    (launch as jest.Mock).mockClear()
})

test('newSession', async () => {
    const client = await DevTools.newSession({
        logLevel: 'trace',
        capabilities: {
            browserName: 'chrome'
        }
    })

    /**
     * don't include platform specific information in snapshot
     */
    delete client.options.capabilities.platformName
    delete client.options.capabilities.platformVersion

    expect(client.options.capabilities).toMatchSnapshot()
    expect(client.options.requestedCapabilities).toMatchSnapshot()
    expect(client.isDevTools).toBe(true)
    expect(client.isChrome).toBe(true)
    expect(client.isW3C).toBe(true)
    expect(launch).toBeCalledTimes(1)
})

test('reloadSession', async () => {
    const client = await DevTools.newSession({
        logLevel: 'trace',
        capabilities: {
            browserName: 'chrome'
        }
    })
    const origSessionId = client.sessionId
    const newClient = await DevTools.reloadSession(client)
    expect(origSessionId).toBe(newClient)
})

test('attachSession', async () => {
    const client = await DevTools.newSession({
        logLevel: 'trace',
        capabilities: {
            browserName: 'chrome'
        }
    })
    const otherClient = await DevTools.attachToSession(client)

    /**
     * don't include platform specific information in snapshot
     */
    delete otherClient.options.capabilities.platformName
    delete otherClient.options.capabilities.platformVersion

    expect(otherClient.capabilities).toMatchSnapshot()
    expect(otherClient.requestedCapabilities).toMatchSnapshot()
    expect(otherClient.isDevTools).toBe(true)
    expect(otherClient.isChrome).toBe(true)
    expect(otherClient.isW3C).toBe(true)
    expect(launch).toBeCalledTimes(2)
})
