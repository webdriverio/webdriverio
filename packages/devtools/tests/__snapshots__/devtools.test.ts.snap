// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`attachSession 1`] = `
Object {
  "browserName": "Chrome",
  "browserVersion": "80.0.3987.149",
  "goog:chromeOptions": Object {
    "debuggerAddress": "localhost:49375",
  },
}
`;

exports[`attachSession 2`] = `
Object {
  "browserName": "chrome",
}
`;

exports[`newSession 1`] = `
Object {
  "browserName": "Chrome",
  "browserVersion": "80.0.3987.149",
  "goog:chromeOptions": Object {
    "debuggerAddress": "localhost:49375",
  },
}
`;

exports[`newSession 2`] = `
Object {
  "browserName": "chrome",
}
`;
